Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -309,12 +312,43 @@ bool ShouldRebuild()
                 context);
         }
 
+        private async Task<Document> GetDocumentWithCodeAsync(string code)
+        {
+            if (ScriptState == null)
+            {
+                ScriptState = await CSharpScript.RunAsync(string.Empty, ScriptOptions);
```

Comment:
Let's move this lazy initialization into a single location rather than let it be repeated. Ideally, an external caller should never see a null `ScriptState`. There's a skipped test for this somewhere.

Reply:
The best I can think of is to turn this:

``` C#
public ScriptState ScriptState { get; private set; }
```

into this:

``` C#
private ScriptState _scriptState;
public ScriptState ScriptState
{
    get
    {
        if (_scriptState == null)
        {
            _scriptState = CSharpScript.RunAsync(string.Empty, ScriptOptions).Result;
        }

        return _scriptState;
    }
    private set
    {
        _scriptState = value;
    }
}
```

If we're willing to remove the `ScriptState` property and turn it into `Task<ScriptState> GetScriptState()` then we can get rid of the `.Result` call in the `get` method, but if we want to keep the property the only way I can think of is to keep it synchronous; that's the only way to guarantee that a null property isn't accessed too early.

I searched through the tests and couldn't find one that was disabled due to a `null` ScriptState, but I found the `null` error from the LSP test, so maybe that's good enough.

Reply:
Looks like the test got deleted at some point.

One approach to avoiding the need to call `.Result` is to call `DeferCommand`. The command will get executed later in an async context.

Reply:
We can't wrap it in `DeferCommand` because the hover method can happen outside of kernel commands; it's just a method on `CSharpKernel`, it needs to be callable from anywhere that a C# kernel exists.

Reply:
I experimented with `Task<ScriptState> GetScriptState()` but that sets off a cascade of `async` that wreaks havoc all over the place, so I added a commit with the `.Result` I mentioned above to see how it feels.